### PR TITLE
Issue #123 - Other Resource Page Select 100% Width Bug

### LINKED ISSUE #123: Other Resources Page - 'Choose a Resource' Dropdown 100% Width after Page Resize
On the 'Other Resources' Page, if after page load the window is resized in any way (horizontal or vertical) the 'Choose a Resource' dropdown will expand to 100% of the page width when clicked. See screenshot.

image

Occurred on Fedora 23 in Chrome & Firefox

### PR DESCRIPTION
**Pull request by MarkDeMaria**
Fixes #123 

Moves ui-select tag inside of a constraining div to ensure it never goes beyond its intended bounds.

@benjaminapetersen

### CODE REVIEW

**Comment by spadgett** [comment]
@MarkDeMaria You should use `git rebase` instead of `git pull` or `git merge` to avoid merge commits.

File: app/views/other-resources.html [hunk 1]
```diff
@@ -11,14 +11,16 @@
           </div>
           <alerts alerts="alerts"></alerts>
           <div class="data-toolbar">
-            <ui-select ng-model="kindSelector.selected" theme="bootstrap" search-enabled="true" ng-disabled="kindSelector.disabled" title="Choose a resource" style="width: 250px;">
-              <ui-select-match placeholder="Choose a resource to list...">{{$select.selected.kind | humanizeKind : true}}</ui-select-match>
-              <ui-select-choices repeat="kind in kinds | filter : {kind: $select.search} : matchKind | orderBy : 'kind'">
-                <div ng-bind-html="(kind.kind | humanizeKind : true) | highlight: $select.search"></div>
-              </ui-select-choices>
-            </ui-select>
+            <div style="width: 250px;">
```
Comment (spadgett) [inline app/views/other-resources.html]: I realize this was an inline style before, but let's make it a CSS class.

File: dist/scripts/vendor.js [hunk 2]
```diff
@@ -39719,12 +39719,13 @@ prevent:"=",
...
 screenKeys:"=",
-autofocus:"=?"
+autofocus:"=?",
+status:"="
 },
 link:function(g, h, i) {
 function j() {
 function a(a) {
-!a && j && (a = "Could not connect to the container. Do you have sufficient privileges?"), a || (a = "disconnected"), j || (a = "\r\n" + a), q.write("[31m" + a + "[m\r\n"), g.$apply(k);
+!a && j && (a = "Could not connect to the container. Do you have sufficient privileges?"), a || (a = "disconnected"), j || (a = "\r\n" + a), q.write("[31m" + a + "[m\r\n"), g.status = "disconnected", g.$apply(k);
```
Comment (spadgett) [inline dist/scripts/vendor.js]: You have some unrelated changes in vendor.js. You probably need to clean/install dist and grunt build again.

**Comment by benjaminapetersen** [comment]
@MarkDeMaria Looks like 1 more try on that rebase, this got a couple extra commits in it. Lemme know if a refresher needed!

**Comment by MarkDeMaria** [comment]
@spadgett @jwforres @benjaminapetersen 

Fixed.

File: app/views/other-resources.html [hunk 3]
```diff
@@ -11,14 +11,16 @@
           </div>
           <alerts alerts="alerts"></alerts>
           <div class="data-toolbar">
-            <ui-select ng-model="kindSelector.selected" theme="bootstrap" search-enabled="true" ng-disabled="kindSelector.disabled" title="Choose a resource" style="width: 250px;">
-              <ui-select-match placeholder="Choose a resource to list...">{{$select.selected.kind | humanizeKind : true}}</ui-select-match>
-              <ui-select-choices repeat="kind in kinds | filter : {kind: $select.search} : matchKind | orderBy : 'kind'">
-                <div ng-bind-html="(kind.kind | humanizeKind : true) | highlight: $select.search"></div>
-              </ui-select-choices>
-            </ui-select>
+            <div class="col-xs-4 col-sm-4 col-md-2 pad-left-none pad-right-none">
```
Comment (spadgett) [inline app/views/other-resources.html]: How does this look at mobile screen widths (~350px or so)? `col-xs-4` sounds really small.

Typically the `col-*` classes go inside a `<div class="row">`, and you would want to use them for other items in the toolbar like the filter. I think instead of `col-*`, I'd just create a new class that sets the widths and padding for this control.

Comment (benjaminapetersen) [inline app/views/other-resources.html]: I think this works here, but just a quick note. I tend to avoid mixing bootstrap classes w/classes that also affect margin/padding/float on the same DOM node.  The reason being that its helpful to keep bootstraps class behavior predictable for mobile resize, etc.

A quick example:

```html
<div class="row">
  <div class="col-md-5">
     <div class="my-class">Hello</div> <!-- nested 'my-class' can have margin,padding,etc w/o breaking the grid -->
  </div>
  <div class="col-md-7">
    <div class="my-other-class">World</div> <!-- same -->
  </div>
</div>
```

**Comment by MarkDeMaria** [comment]
@spadgett @benjaminapetersen 

Had a conversation with @rhamilto regarding this issue. This is a bug in angular-ui-select (see angular-ui/ui-select#1678). The fix is described at the bottom of that issue thread.

File: app/styles/_core.less [hunk 4]
```diff
@@ -79,6 +79,10 @@
 
 /* Fix some styles with the angular-ui-select bootstrap theme that isn't right when combined with all of our styles */
 .ui-select-bootstrap {
+  .ui-select-search {
+    width: 100% !important;
```
Comment (spadgett) [inline app/styles/_core.less]: Add a comment linking to the ui-select issue.

File: app/views/other-resources.html [hunk 5]
```diff
@@ -11,14 +11,14 @@
           </div>
           <alerts alerts="alerts"></alerts>
           <div class="data-toolbar">
-            <ui-select ng-model="kindSelector.selected" theme="bootstrap" search-enabled="true" ng-disabled="kindSelector.disabled" title="Choose a resource" style="width: 250px;">
-              <ui-select-match placeholder="Choose a resource to list...">{{$select.selected.kind | humanizeKind : true}}</ui-select-match>
-              <ui-select-choices repeat="kind in kinds | filter : {kind: $select.search} : matchKind | orderBy : 'kind'">
-                <div ng-bind-html="(kind.kind | humanizeKind : true) | highlight: $select.search"></div>
-              </ui-select-choices>
-            </ui-select>
+              <ui-select ng-model="kindSelector.selected" theme="bootstrap" search-enabled="true" ng-disabled="kindSelector.disabled" title="Choose a resource">
```
Comment (spadgett) [inline app/views/other-resources.html]: Indentation

**Comment by spadgett** [comment]
Thanks @MarkDeMaria, this fix looks simpler.

**Comment by spadgett** [comment]
And thanks @rhamilto

**Comment by MarkDeMaria** [comment]
@spadgett Ready for merge

File: app/styles/_core.less [hunk 6]
```diff
@@ -79,6 +79,11 @@
 
 /* Fix some styles with the angular-ui-select bootstrap theme that isn't right when combined with all of our styles */
 .ui-select-bootstrap {
+  /* Fixes ui-select's input box extending to 100% page width if the page has been resized after page-load */
```
Comment (spadgett) [inline app/styles/_core.less]: I'd add a link to the GitHub issue as well. Then we know when the issue is fixed and can take out the workaround.

**Comment by MarkDeMaria** [comment]
@spadgett Done

File: app/views/other-resources.html [hunk 7]
```diff
@@ -11,14 +11,14 @@
           </div>
           <alerts alerts="alerts"></alerts>
           <div class="data-toolbar">
-            <ui-select ng-model="kindSelector.selected" theme="bootstrap" search-enabled="true" ng-disabled="kindSelector.disabled" title="Choose a resource" style="width: 250px;">
+            <ui-select ng-model="kindSelector.selected" theme="bootstrap" search-enabled="true" ng-disabled="kindSelector.disabled" title="Choose a resource">
```
Comment (spadgett) [inline app/views/other-resources.html]: Sorry just noticed: it looks like we lost the width style?

**Comment by MarkDeMaria** [comment]
@spadgett My mistake. Re-added.

File: app/styles/_core.less [hunk 8]
```diff
@@ -79,6 +79,13 @@
 
 /* Fix some styles with the angular-ui-select bootstrap theme that isn't right when combined with all of our styles */
 .ui-select-bootstrap {
+  /* Fixes ui-select's input box extending to 100% page width if the page has been resized after page-load
```
Comment (benjaminapetersen) [inline app/styles/_core.less]: Nice fix. Do you have a link to an issue with ui-select instead? It might be more helpful. If they ever fix it, we can eliminate this little hack.

**Comment by MarkDeMaria** [comment]
@benjaminapetersen Done.

**Comment by benjaminapetersen** [comment]
Cool. Only thing left is @spadgett's comment about unrelated changes in vendor.js.  prob just a `hack/clean-deps.sh`, `hack/install-deps.sh` and a new `grunt build`.

**Comment by MarkDeMaria** [comment]
@benjaminapetersen @spadgett Completed clean/install and grunt build.

**Comment by spadgett** [comment]
[merge]
